File: UnityProject/Assets/Scripts/Data/FBattleFieldDataManager.cs
using System.Collections.Generic;

public class FBattleFieldData
{
    public readonly int id;
    public readonly int price;
    public readonly string name;
    public readonly string skinImagePath;
    public readonly string battlefieldPrefab;
    
    public FBattleFieldData(FDataNode InNode)
    {
        id = InNode.GetIntAttr("id");
        price = InNode.GetIntAttr("price");
        name = InNode.GetStringAttr("name");
        skinImagePath = InNode.GetStringAttr("skinImage");
        battlefieldPrefab = InNode.GetStringAttr("battlefieldPrefab");
    }
}

public class FBattleFieldDataManager : FNonObjectSingleton<FBattleFieldDataManager>
{
    Dictionary<int, FBattleFieldData> battleFieldDataMap = new Dictionary<int, FBattleFieldData>();

    public void Initialize()
    {
        List<FDataNode> dataNodeList = FDataCenter.Instance.GetDataNodesWithQuery("BattleFieldList.BattleField");
        foreach (FDataNode node in dataNodeList)
        {
            FBattleFieldData newData = new FBattleFieldData(node);
            battleFieldDataMap.Add(newData.id, newData);
        }
    }

    public delegate void ForeachBattleFieldDataFunc(FBattleFieldData InData);
    public void ForeachBattleFieldData(ForeachBattleFieldDataFunc InFunc)
    {
        foreach(FBattleFieldData data in battleFieldDataMap.Values)
        {
            InFunc(data);
        }
    }

    public FBattleFieldData FindBattleFieldData(int InID)
    {
        if (battleFieldDataMap.ContainsKey(InID))
            return battleFieldDataMap[InID];

        return null;
    }
}

File: Server/Server/Server/FBattlefieldDataManager.cpp
#include "stdafx.h"
#include "FBattlefieldDataManager.h"

void FBattlefieldDataManager::Initialize()
{
    vector<const FDataNode*> battlefieldNodeList = FDataCenter::GetInstance()->FindNodes("BattleFieldList.BattleField");
    for (const FDataNode* battlefieldNode : battlefieldNodeList)
    {
        FBattlefieldData battlefieldData;
        battlefieldData.id = battlefieldNode->GetIntAttr("id");
        battlefieldData.price = battlefieldNode->GetIntAttr("price");

        m_BattlefieldDataMap.insert(pair<int, FBattlefieldData>(battlefieldData.id, battlefieldData));
    }
}

const FBattlefieldData* FBattlefieldDataManager::FindBattlefieldData(int InID) const
{
    auto iter = m_BattlefieldDataMap.find(InID);
    if (iter != m_BattlefieldDataMap.end())
        return &iter->second;

    return nullptr;
}
